packages/provider/src/json-rpc-provider.ts

```diff
@@ -42,22 +42,36 @@ export class JsonRpcProvider implements AbstractProvider {
     body,
     rpcUrl,
     timeout = DEFAULT_TIMEOUT,
+    retryAttempts = 0,
+    attemptsPerformed = 1
   }: {
     route: V1RpcRoutes
     body: any
     rpcUrl?: string
-    timeout?: number
+    timeout?: number,
+    retryAttempts?: number,
+    attemptsPerformed?: number
   }): Promise<Response> {
+    const shouldRetryOnFailure = attemptsPerformed < retryAttempts
```

Comment:
Hmm, not sure if this is correct. If

- `retryAttempts = 1`
- `attemptsPerformed = 1` (default value)

The function will not retry, as `attemptsPerformed === retryAttempts`. So the param won't really work like it's supposed to—for having 1 retry, you'll need to put `2` on `retryAttempts`. `attemptsPerformed` should start on `0` I believe, and probably should be called `retriesPerformed`.